common/src/main/java/earth/terrarium/olympus/client/components/compound/radio/RadioBuilder.java

```diff
@@ -0,0 +1,108 @@
+package earth.terrarium.olympus.client.components.compound.radio;
+
+import earth.terrarium.olympus.client.components.Widgets;
+import earth.terrarium.olympus.client.components.base.renderer.WidgetRenderer;
+import earth.terrarium.olympus.client.components.base.renderer.WidgetRendererContext;
+import earth.terrarium.olympus.client.components.compound.CompoundWidget;
+import earth.terrarium.olympus.client.components.renderers.WidgetRenderers;
+import earth.terrarium.olympus.client.layouts.Layouts;
+import earth.terrarium.olympus.client.ui.UIConstants;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.components.WidgetSprites;
+import net.minecraft.network.chat.CommonComponents;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class RadioBuilder<T> {
+    private final List<T> options = new ArrayList<>();
+    private final RadioState<T> state;
+    private BiFunction<T, Boolean, WidgetRenderer<AbstractWidget>> entryRenderer = (ignored, depressed) -> WidgetRenderers.text(CommonComponents.ELLIPSIS);
+    private Function<T, WidgetSprites> entrySprites = (ignored) -> UIConstants.BUTTON;
+    private Consumer<T> action = t -> {};
+
+    private int width = 120;
+    private int height = 20;
+    private int gap = 0;
+
+    private boolean preventDeselection = false;
```

Comment:
this should be true by default and should be changed to something like `allowSelecting`